### DISCUSSION TITLE
yqlib adds a trailing \n to scalar outputs

### DISCUSSION THREAD

**Discussion by ElanHasson** [mikefarah/yq#1709] [Q&A]
I've tried a few different settings on the preferences to get the `\n` to not be appended to the end with no luck.

I also tried the Json encoder as well. Not sure what to do to get it to not append it.

Not sure if this is the same as described here in : https://github.com/mikefarah/yq/discussions/1641
Thanks!

```go
        input := `{"foo":1}`
        expression := ".foo"
	prefs := yqlib.ConfiguredYamlPreferences
	prefs.UnwrapScalar = false
	encoder := yqlib.NewYamlEncoder(0, false, prefs)
	decoder := yqlib.NewJSONDecoder()

	y, err := json.Marshal(input)
	if err != nil {
		return "", err
	}

	result, err := yqlib.NewStringEvaluator().Evaluate(expression, input,  encoder, decoder)
```

**Actual behavior**

```yaml
result = "1\n"
```

**Expected behavior**

```yaml
result = "1"
```

**Comment by mikefarah** [comment]
Yeah it does that on purpose - as that's what JQ also does. 

Can you provide more context on what you're trying to do - as far as I know *sh clis will chomp the newline when you capture the value in a variable anyway..

**Reply by ElanHasson** [reply]
I'm using YQ package in a go application to serve as a expression evaluation tool against yaml.

And I'm wondering if I should just look for the presence of the \n and trim? Or if the encoder can do it itself?
